utils/land-patterns.stanza

```diff
@@ -9,23 +9,147 @@ defpackage ocdb/land-patterns :
+  val [length, width_, rot] = normalize-rotation(width, height)
+  normalize-loc(Rectangle(width, height), anchor,rot)
+
+; Create a capsule with normalized rotation and position.
+defn capsule (anchor:Anchor, width:Double, height:Double):
+  val [length, width_, rot] = normalize-rotation(width, height)
+  normalize-loc(Capsule(length, width_), anchor, rot)
+
+; Create a rounded rectangle with normalized rotation and position.
+defn round-rect (anchor:Anchor, width:Double, height:Double, r:Double):
+  val [length, width_, rot] = normalize-rotation(width, height)
+  normalize-loc(RoundedRectangle(length, width_ r), anchor, rot)
+  
+; Create a chamfered rectangle with normalized rotation and position.
+defn chamfered (anchor:Anchor, width:Double, height:Double, r:Double):
+  val [length, width_, rot] = normalize-rotation(width, height)
+  normalize-loc(ChamferedRectangle(length, width_, r), anchor, rot)
+
+;==============================================================================
+;=========================== Geometric Helpers ================================
+;==============================================================================
+; Create a rectangle out of lines
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double, line-w:Double) :
+  val w2 = w * 0.5
+  val h2 = h * 0.5
+  Line(line-w, [
+    Point(xc - w2, yc - h2)
+    Point(xc + w2, yc - h2)
+    Point(xc + w2, yc + h2)
+    Point(xc - w2, yc + h2)
+    Point(xc - w2, yc - h2)])
 
-; ====== Land pattern utilities ===============================================
+; Create a rectangle with width, height, center x/y and default line width
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double) :
+  LineRectangle(w, h, xc, yc, MIN-SILKSCREEN-WIDTH)
+
+; Create a rectangle with width and height and default x/y position
+public defn LineRectangle (w:Double, h:Double) :
+  LineRectangle(w, h, 0.0, 0.0)
+
+; Create a rectangle from dims structure
+public defn LineRectangle (d:Dims): 
+  LineRectangle(x(d), y(d))
+
+; Create a rectangle around another shape
+public defn outline-bounding-box (s:Shape):
+  LineRectangle(dims(s))
+
+; Create a "+" symbol with a fixed length at a center point
+public defn PlusSymbol (center:Point, l:Double) -> Shape:
+  Union $ [
+    Line(MIN-SILKSCREEN-WIDTH, [center + Point(-0.5 * l, 0.0), center + Point(0.5 * l, 0.0)])
+    Line(MIN-SILKSCREEN-WIDTH, [center + Point(0.0, -0.5 * l), center + Point(0.0, 0.5 * l)])
+  ]
+
+; Create a 1mm x 1mm plus symbol at Point(x, y)
+public defn PlusSymbol (x:Double, y:Double) -> Shape:
+  PlusSymbol(Point(x, y), 1.0)
+
+; Create a row of poses
+public defn pose-row (origin:Pose, count:Int, pitch:Double) -> Seqable<Pose>:
+  for n in 0 to count seq:
+    Pose(center(origin) + Point(pitch, 0.0), angle(origin), flip-x?(origin))
+
+; Create a column of poses
+public defn pose-column (origin:Pose, count:Int, pitch:Double) -> Seqable<Pose>:
```

Comment:
1) I don't quite know what this is supposed to create. I originally thought it was supposed to create a bunch of poses that were offset from each other by `pitch` in the y direction. This is how `col-locs` works. If this is indeed what it is supposed to do, please use `col-locs` instead to reduce duplication.

2) Perhaps I misunderstood the intent of this function. It doesn't use the `pitch` argument anywhere. And you create a sequence of identical poses because the `n` loop variable is not used anywhere.

Reply:
Removed.